jujugui/static/gui/src/app/components/lightbox/lightbox.js

```diff
@@ -1,35 +1,155 @@
-/* Copyright (C) 2015 Canonical Ltd. */
-
+/* Copyright (C) 2017 Canonical Ltd. */
 'use strict';
 
 const PropTypes = require('prop-types');
 const React = require('react');
+const classNames = require('classnames');
 
 const SvgIcon = require('../svg-icon/svg-icon');
 
 /**
   Renders a modal like overlay with a darkened background.
   A caption is displayed in white below the content if present.
 */
-class Lightbox extends React.PureComponent {
+class Lightbox extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      activeSlide: 0,
+      lastSlide: null
+    };
+  }
+
+  componentWillMount() {
+    if (this.props.children.map) {
+      this.setState({
+        lastSlide: this.props.children.length - 1
```

Comment:
It seems the rest of the component supports a single child being passed....but `lastSlide` will be `null` if that's so, breaking the component.

Reply:
I don't think this is true. If there is only one child, everything that uses `lastSlide` is skipped.